src/table/paginator.scss

```diff
@@ -57,28 +57,34 @@
     color: t(iui-text-color-muted);
   }
 
-  &.iui-small {
+  &-small {
     width: $iui-component-height-small;
   }
 }
 
 @mixin iui-paginator-pages-group {
+  @include iui-button-group;
   margin: 0 $iui-s;
 
   > * {
-    margin-right: 0; // unset negative margin from button-group
-  }
-
-  .iui-paginator-page-button {
-    width: $iui-component-height;
-
-    &.iui-small {
-      width: $iui-component-height-small;
+    &:not(:last-child) {
```

Comment:
I will create a separate PR to lower specificity here after this is merged

![image](https://user-images.githubusercontent.com/9084735/147958959-9496cf86-8b04-4719-a644-b04d47497a7a.png)
![image](https://user-images.githubusercontent.com/9084735/147958963-0c924569-0ffc-4f02-aad5-612496720f5d.png)